### PR TITLE
Using importHelpers instead of noEmitHelpers

### PR DESCRIPTION
**Pull request by rorticus**
Using `importHelpers` instead of `noEmitHelpers`. Helpers will be provided by `tslib` but augmented by `@dojo/shim/main`.

### CODE REVIEW

**Review by bryanforbes — APPROVED** [review]

**Review by bryanforbes — CHANGES_REQUESTED** [review]
Can you go through and make sure that where we're using `for...of` that it's not on arrays? Or, if it's on an array, we treat them differently because of https://github.com/Microsoft/TypeScript/issues/19933 .

**Comment by rorticus** [comment]
@bryanforbes I replaced the array-only usages of `for..of` with `forEach`

File: src/lang.ts [hunk 1]
```diff
@@ -51,13 +52,15 @@ function _mixin<T extends {}, U extends {}>(kwArgs: MixinArgs<T, U>): T&U {
...
+	for (let i = 0; i < kwArgs.sources.length; i++) {
+		const source = kwArgs.sources[ i ];
+
 		if (source === null || source === undefined) {
 			continue;
 		}
 		for (let key in source) {
 			if (inherited || hasOwnProperty.call(source, key)) {
-				let value: any = source[key];
+				let value: any = source[ key ];
```
Comment (bryanforbes) [inline src/lang.ts]: Why are you adding spaces around `key`?

**Comment by rorticus** [comment]
@bryanforbes Webstorm was overzealous. I've put back (most) of the style changes. I left ones alone that were fixing previous style errors because of reasons.

File: src/load/util.ts [hunk 2]
```diff
@@ -36,7 +36,16 @@ export function isPlugin(value: any): value is LoadPlugin<any> {
...
+	if (isArrayLike(modules)) {
+		let processedModules: any[] = [];
+
+		for (let i = 0; i < modules.length; i++) {
+			const module = modules[i];
+			processedModules.push((module.__esModule && module.default) ? module.default : module);
+		}
+
+		return processedModules;
+	} else if (isIterable(modules)) {
```
Comment (bryanforbes) [inline src/load/util.ts:48]: Should the `else if` be on a new line?

**Review by bryanforbes — CHANGES_REQUESTED** [review]

File: src/load/util.ts (hunk 2 above)
Comment (rorticus) [reply src/load/util.ts:48]: it is now!

**Review by bryanforbes — APPROVED** [review]

**Pull request merged**